### PR TITLE
Acceptance tests for TerminalLogger

### PR DESCRIPTION
**Pull request by rainersigwald**
Convert LiveLogger tests to use Verify
    
* Bump Xunit version to one compatible with Verify 19.14.1
* Remove MockTerminal in favor of internal fields on Terminal itself.
* Reference Verify library for acceptance testing.
* Convert existing tests to Verify.

### CODE REVIEW

File: src/MSBuild/LiveLogger/Terminal.cs [hunk 1]
```diff
@@ -29,18 +30,49 @@ internal sealed class Terminal : ITerminal
...
+
     /// <inheritdoc/>
-    public int Height => Console.BufferHeight;
+    public int Height
+    {
+        get
+        {
+            if (Console.IsOutputRedirected)
+            {
+                return int.MaxValue;
```
Comment (rainersigwald) [inline src/MSBuild/LiveLogger/Terminal.cs:42]: Should we default to "infinite", like I've done here, or to "a reasonable size like 80×20"?

Comment (ladipro) [reply src/MSBuild/LiveLogger/Terminal.cs:42]: For redirected output it makes sense to use infinite. For tests I think we may want to use reasonable (as well as unreasonable!) values. I would probably add `int?` parameters to the new constructor and default to infinite if null.

Comment (rainersigwald) [reply src/MSBuild/LiveLogger/Terminal.cs:42]: I agree--we should manipulate that to test the selective-truncation stuff. I'll leave that for later.

**Comment by rainersigwald** [comment]
> It's failing because the test tool long enough that the refresher thread pushed out a status! Adding a manual mode and using that in the tests (well technically none of these tests need it yet).

It wasn't _just_ that--the time in `(0.1s)` can also change. Running tests again to validate my scrubber.

File: src/MSBuild/LiveLogger/Terminal.cs [hunk 2]
```diff
@@ -29,18 +30,51 @@ internal sealed class Terminal : ITerminal
...
+
+    private const int BigUnknownDimension = 2 << 23;
+
     /// <inheritdoc/>
-    public int Height => Console.BufferHeight;
+    public int Height
+    {
+        get
+        {
+            if (Console.IsOutputRedirected)
```
Comment (Forgind) [inline src/MSBuild/LiveLogger/Terminal.cs:42]: Just to be pedantic, you could theoretically redirect output to StdOut, right? (Not that that does anything) I'm wondering if there are ways to figure out the right dimensions in certain special cases. If you don't want to bother, I don't think it's actually important.

File: src/MSBuild.UnitTests/LiveLogger_Tests.cs [hunk 3]
```diff
@@ -4,35 +4,59 @@
...
+
+        private readonly Terminal _mockTerminal;
         private readonly LiveLogger _liveLogger;
 
         private readonly DateTime _buildStartTime = new DateTime(2023, 3, 30, 16, 30, 0);
         private readonly DateTime _buildFinishTime = new DateTime(2023, 3, 30, 16, 30, 5);
 
+        private VerifySettings _settings = new();
+
+        private static Regex _elapsedTime = new(@"\(\d+\.\ds\)", RegexOptions.Compiled);
```
Comment (ladipro) [inline src/MSBuild.UnitTests/LiveLogger_Tests.cs:40]: Ok to hard-code dot as the decimal separator? Can the test be running under locales that use a different character?

**Review by ladipro — APPROVED** [review]
>> It's failing because the test tool long enough that the refresher thread pushed out a status! Adding a manual mode and using that in the tests (well technically none of these tests need it yet).

> It wasn't _just_ that--the time in `(0.1s)` can also change. Running tests again to validate my scrubber.

I wish there was an easy way to mock `DateTime.Now`.

File: src/MSBuild.UnitTests/LiveLogger_Tests.cs [hunk 4]
```diff
@@ -4,35 +4,59 @@
...
 {
+    [UsesVerify]
     public class LiveLogger_Tests : IEventSource, IDisposable
     {
         private const int _nodeCount = 8;
-        private const int _terminalWidth = 80;
-        private const int _terminalHeight = 40;
         private const string _eventSender = "Test";
-        private const string _projectFile = @"C:\src\project.proj";
+        private readonly string _projectFile = NativeMethods.IsUnixLike ? "/src/project.proj" : @"C:\src\project.proj";
```
[the remark this answers is not in the document]
Comment (rainersigwald) [reply src/MSBuild.UnitTests/LiveLogger_Tests.cs:28]: I expect this to change radically in the next weeks as we expand tests (to include multiple-project scenarios) so I'm not going to add a comment now.

File: src/MSBuild.UnitTests/LiveLogger_Tests.cs (hunk 3 above)
Comment (rainersigwald) [reply src/MSBuild.UnitTests/LiveLogger_Tests.cs:40]: I'm going to push a change that makes this somewhat more robust but to get it really right I think we'll have to refactor how our localized resources work. Right now it doesn't look like there's anything we could extract that would localize the `s`.

File: eng/dependabot/Packages.props [hunk 5]
```diff
@@ -48,6 +48,9 @@
 
     <PackageVersion Include="System.Security.Cryptography.X509Certificates" Version="4.3.2" />
     <PackageVersion Update="System.Security.Cryptography.X509Certificates" Condition="'$(SystemSecurityCryptographyX509CertificatesVersion)' != ''" Version="$(SystemSecurityCryptographyX509CertificatesVersion)" />
+
+    <PackageVersion Include="Verify.Xunit" Version="19.14.1" />
+    <PackageVersion Update="Verify.XUnit" Condition="'$(VerifyXUnitVersion)' != ''" Version="$(VerifyXUnitVersion)" />
```
[the remark this answers is not in the document]
Comment (rainersigwald) [reply eng/dependabot/Packages.props:53]: It's an external project that is already used in several other .NET repos, including the SDK. https://github.com/dotnet/sdk/blob/e425d8ca99e8b306f94a76a75e8c1cac269d1b49/eng/dependabot/Packages.props#LL7C5-L7C64

File: src/MSBuild.UnitTests/Snapshots/LiveLogger_Tests.PrintsBuildSummary_Succeeded.verified.txt [hunk 6]
```diff
@@ -0,0 +1,3 @@
+﻿[?25l[1F
```
[the remark this answers is not in the document]
Comment (rainersigwald) [reply src/MSBuild.UnitTests/Snapshots/LiveLogger_Tests.PrintsBuildSummary_Succeeded.verified.txt:1]: This is a reasonable rendering. The output includes [VT codes](https://learn.microsoft.com/windows/console/console-virtual-terminal-sequences) that, when interpreted by a terminal emulator, apply formatting or other properties to the output, like color or overwriting an already-printed line.

You can render the completed by piping it through a terminal emulator, for example:

image

File: src/MSBuild/LiveLogger/Terminal.cs (hunk 2 above)
Comment (rainersigwald) [reply src/MSBuild/LiveLogger/Terminal.cs:42]: I don't think that's interesting enough to jump through hoops for at this point.

File: eng/dependabot/Packages.props (hunk 5 above)
[the remark this answers is not in the document]
Comment (JanKrivanek) [reply eng/dependabot/Packages.props:53]: Very favorit OSS within our repos. Please remember to share some love back during the June FOSS voting! (https://github.com/microsoft/foss-fund).

**Review by JanKrivanek — APPROVED** [review]
